### ISSUE TITLE
Caching Buffers?

### ISSUE THREAD

**Issue by zz85** [oampo/Audiolet#19]
Hi, just wondering have anyone done any caching with audiolet?

For example, if you create a 1second sine wave, and were to play it back multiple times, would it possible to create a pre-calculated audio object? With this audio object, it could be more efficient to playback the object anytime you need without generating/calculating the buffers again.

**Comment by oampo** [comment]
I haven't, but it should be possible as part of the planned BufferRecorder node - I talk a little bit about it in issue #11.  The basic idea is that you could insert a BufferRecorder node somewhere in your processing chain and push a certain number of samples into it.  When you have enough cached you would then disconnect everything from the BufferRecorder, and replace it with a looping BufferPlayer using the AudioletBuffer which you have been adding data to.

I haven't had time to have a proper go at creating the BufferRecorder but in theory it's pretty simple - just a buffer and a gate would do to start with.  If you fancy having a go at it then I'll be happy to take a look at a pull request, and if you have any questions or issues then just post them up here.

**Comment by zz85** [comment]
i'm new to all these audio stuff, but from what it seems needs to be done is to generate the waveforms and turn them into pcm wave, and then base64 encode them to be passed to an audio object? that's probably what sink.record() seems to be doing.

**Comment by oampo** [comment]
Yeah, that would be a slightly different approach to achieving the same thing - each way of doing it has it's advantages and disadvantages.  If you don't want to do any further processing on the audio then having it as an audio object would make sense as lots of the processing gets offloaded away from the JavaScript.  If you still want to process the audio, just with some intensive processing cached then keeping everything with Audiolet as buffers makes more sense I think.
